module/Activity/src/Activity/Controller/AdminController.php

```diff
@@ -88,6 +102,101 @@ public function exportPdfAction()
         return $pdf;
     }
 
+    public function externalSignupAction()
+    {
+        $id = (int) $this->params('id');
+        $activityService = $this->getServiceLocator()->get('activity_service_activity');
+        $queryService = $this->getServiceLocator()->get('activity_service_activityQuery');
+        $signupService = $this->getServiceLocator()->get('activity_service_signup');
+
+        $activity = $queryService->getActivity($id);
+
+        $translator = $activityService->getTranslator();
+
+        //Assure the form is used
+        if (!$this->getRequest()->isPost()){
+            $error = $translator->translate('Use the form to subscribe');
+            $this->redirectSignupRequest($id, false, $error);
+            return;
+        }
+
+        $form = $signupService->getForm($activity->getFields(), true);
+        $form->setData($this->getRequest()->getPost());
+
+        //Assure the form is valid
+        if (!$form->isValid()){
+            $error = $translator->translate('Invalid form');
+            $this->redirectSignupRequest($id, false, $error);
+            return;
+        }
+
+        $formData = $form->getData(\Zend\Form\FormInterface::VALUES_AS_ARRAY);
+        $fullName = $formData['fullName'];
+        unset($formData['fullName']);
+        $email = $formData['email'];
+        unset($formData['email']);
```

Comment:
This array, apart from the keys `'fullName'` and `'email'`, contains keys referring to values from additional signup fields. Those need to be passed, but the email and fullName must be passed separately. An alternative is to remove these values in the service, but they have to be removed somewhere.